californium-core/src/main/java/org/eclipse/californium/core/network/MessageExchangeStore.java

```diff
@@ -181,6 +184,15 @@
 	boolean isEmpty();
 
 	/**
+	 * Checks if there are any exchanges currently being registered in this
+	 * store and dump these exchanges. Intended to be used for more details in
+	 * log when testing.
+	 * 
+	 * @return {@code true} if there no exchanges registered.
+	 */
+	boolean isEmptyWithDump();
```

Comment:
Why ? It seems to me that this is for unit testing purpose only.  And AFAIK, we have only 1 implementation to test and I bet we will not add a new one at short/mid-term.